src/notebooks/controllers/kernelSource/notebookKernelSourceSelector.ts

```diff
@@ -108,11 +114,21 @@ interface ConnectionQuickPickItem extends QuickPickItem {
 
 // The return type of our multistep selection process
 type MultiStepResult = {
+    notebook: NotebookDocument;
     source?: IContributedKernelFinder;
     connection?: KernelConnectionMetadata;
     disposables: IDisposable[];
 };
 
+function updateQuickPickWithNewItems(
+    quickPick: QuickPick<KernelSourceQuickPickItem | QuickPickItem>,
+    items: (KernelSourceQuickPickItem | QuickPickItem)[],
+    activeItem?: KernelSourceQuickPickItem | QuickPickItem
+) {
+    const activeItems = quickPick.activeItems.length ? [quickPick.activeItems[0]] : activeItem ? [activeItem] : [];
```

Comment:
Is it possible that `quickPick.activeItems` no longer exists in `items`?

Reply:
I did this recalculation of active items in core kernel picker https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/notebook/browser/contrib/editorStatusBar/editorStatusBar.ts#L226-L240